lib/documents.js

```diff
@@ -98,9 +105,34 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
  * @returns {Promise<object>} An object with the registration record.
  */
 export async function register({
-  internalId, externalId, document, recipients, ttl, creator,
+  internalId, externalId, document, recipients, recipientChain, ttl, creator,
   tokenizer, externalIdHash, documentHash, creatorHash, newRegistration
 } = {}) {
+  if(!(recipients || recipientChain)) {
+    throw new BedrockError(
+      'Encryption recipients or recipientChain is required.',
+      'ValidationError', {
+        public: true,
+        httpStatusCode: 400
+      });
+  }
+  if(recipients && recipientChain) {
+    throw new BedrockError(
+      'Either recipients or recipientChain is required, not both.',
+      'ValidationError', {
+        public: true,
+        httpStatusCode: 400
+      });
+  }
+  if(recipientChain && !Array.isArray(recipientChain)) {
+    throw new BedrockError(
+      '"recipientChain" must be an array.',
+      'ValidationError', {
+        public: true,
+        httpStatusCode: 400
+      });
+  }
```

Comment:
So we'd normally just use assert-plus and `TypeError` here. We also need to ensure that at least one recipient has been passed so as to avoid accidentially not encrypting a document at all later (loop doesn't run).

```suggestion
  assert.optionalArrayOfObject(recipients, 'recipients');
  assert.optionalArrayOfArray(recipientChain, 'recipientChain');
  if(recipientChain) {
    if(recipients) {
      throw new TypeError('Only one of "recipients" or "recipientChain" is allowed.');
    }
    recipientChain.forEach(e => assert.arrayOfObject(e));
  } else if(!recipients) {
    throw new TypeError('"recipients" or "recipientChain" is required.');
  }
```

Reply:
Actually -- we may need to test for `.length > 0` on recipients/recipientsChain to ensure some encryption happens. We should have a test for that too (with empty recipients/recipientsChain) to ensure we don't accidentally allow unencrypted docs.

Reply:
Good call, thanks.

Reply:
@dlongley switched to `TypeError`, added additional checks and tests, should be addressed now.